ui/src/app/views/notebook-redirect/component.spec.ts

```diff
@@ -105,140 +104,234 @@ describe('NotebookRedirectComponent', () => {
           snapshot: {
             params: {
               'ns': WorkspaceStubVariables.DEFAULT_WORKSPACE_NS,
-              'wsid': WorkspaceStubVariables.DEFAULT_WORKSPACE_ID
+              'wsid': WorkspaceStubVariables.DEFAULT_WORKSPACE_ID,
+              'nbName': 'blah%20blah'
+            },
+            queryParams: {
+              'kernelType': Kernels.R,
+              'creating': true
             },
             queryParamMap: convertToParamMap({
-              'notebook-name': 'blah',
-              'kernel-type': Kernels.R
-            }),
-            data: {
-              creating: true
-            }
+              'kernelType': Kernels.R,
+              'creating': true
+            })
           }
         }},
-      ]}).compileComponents();
+      ]}).compileComponents().then(() => {
+      fixture = TestBed.createComponent(NotebookRedirectComponent);
+      spyOn(window.history, 'replaceState').and.stub();
+      blockingClusterStub.release();
+      blockingNotebooksStub.release();
+    });
   }));
 
-  function spinnerText() {
-    return fixture.debugElement.query(By.css('.spinner-text'))
-      .nativeElement.textContent;
+  function spinner() {
+    return fixture.debugElement.query(By.css('.spinner-sm'))
+      .nativeElement;
   }
 
-  beforeEach(fakeAsync(() => {
-    fixture = TestBed.createComponent(NotebookRedirectComponent);
-    blockingClusterStub.release();
-    blockingNotebooksStub.release();
-  }));
-
   it('should render', fakeAsync(() => {
     updateAndTick(fixture);
     expect(fixture.componentRef).toBeTruthy();
-
     // Tears down the retrying subscription.
     fixture.destroy();
   }));
 
   it('should redirect', fakeAsync(() => {
     updateAndTick(fixture);
-    expect(fakeWindow.location.href).toEqual('');
+    expect(fixture.componentInstance.leoUrl).toBeFalsy();
     blockingClusterStub.cluster.status = ClusterStatus.Running;
-    tick(10000);
+    tick(100000);
     updateAndTick(fixture);
-
-    expect(fakeWindow.location.href.startsWith(environment.leoApiUrl)).toBeTruthy();
-    expect(fakeWindow.location.href).toContain('/notebooks');
+    expect(fixture.componentInstance.leoUrl).toMatch(environment.leoApiUrl);
   }));
 
-  it('should display "Initializing" until ready', fakeAsync(() => {
+  it('should set spinner "Initializing" until ready', fakeAsync(() => {
     updateAndTick(fixture);
     fixture.detectChanges();
-    expect(spinnerText()).toContain('Initializing');
+    const initBox = fixture.debugElement.queryAll(By.css('#Initializing'))[0];
+    expect(initBox.children[0].nativeElement === spinner());
```

Comment:
These spinner checks feel way too dependent on specific HTML structure.  Just query for existence of `"#initializing .spinner"`

Reply:
So I've tried that strategy before, and had my browser break. I think that was the problem Jackie was running into. Which is why she set it up like this. I just tried again, and ran into the same problem. So leaving as is until we figure out why Jasmine can't seem to handle multiple css identifiers

Reply:
Sounds a lot like this issue: https://precisionmedicineinitiative.atlassian.net/browse/RW-1424 - have you already tried the workaround I listed there?

Reply:
Yes